src/credentials/providers/credentialsProviderId.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const CREDENTIALS_PROVIDER_ID_SEPARATOR = ':'
+
+export interface CredentialsProviderIdComponents {
+    credentialType: string
+    credentialTypeId: string
+}
+
+export function makeCredentialsProviderIdComponents(credentialsProviderId: string): CredentialsProviderIdComponents {
+    const chunks = credentialsProviderId.split(CREDENTIALS_PROVIDER_ID_SEPARATOR)
```

Comment:
I think it is only explicit in the tests, for readability. Anywhere in the toolkit code goes through the make/decode methods.

I initially stopped short of creating a credential provider id object. Since the intent was to support additional data in the future (using the separator), its probably saving future pain to make and pass around an object now instead of a string.

I'll give this a go.